ush/run_mpmd.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+source "${HOMEgfs}/ush/preamble.sh"
+
+cmdfile=${1:?"run_mpmd requires an input file containing commands to execute in MPMD mode"}
+
+# This is a partial implementation for Slurm-only
+# TODO: extend to PBS, etc.
+
+# Slurm requires a counter in front of each line in the script
+# Read the incoming cmdfile and create srun usable cmdfile
+if [[ -s "${DATA:-}/cmdfile_srun" ]]; then rm -f "${DATA:-}/poescript_srun"; fi
+nm=0
+# shellcheck disable=SC2312
+cat "${cmdfile}" | while read line; do
+  echo "${nm} ${line}" >> "${DATA:-}/cmdfile_srun"
+  ((nm=nm+1))
+done
```

Comment:
This should take care of both shellcheck complaints:
```suggestion
while IFS= read -r line; do
  echo "${nm} ${line}" >> "${DATA:-}/cmdfile_srun"
  ((nm=nm+1))
done < ${cmdfile}
```